### PR TITLE
[Merged by Bors] - chore: add porting header to Mathlib.Data.HashMap

### PR DESCRIPTION
**Pull request by kim-em**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Data/HashMap.lean [hunk 1]
```diff
@@ -2,6 +2,15 @@
 Copyright (c) 2022 Scott Morrison. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Scott Morrison
+
+! This file was ported from Lean 3 source module data.set.pairwise.lattice
```
Comment (collares) [inline Mathlib/Data/HashMap.lean:6]: Wrong file?

**Review by jcommelin — APPROVED** [review]
Please fix the filename in the header (and maybe double check the hash?).

bors d+

File: Mathlib/Data/HashMap.lean [hunk 2]
```diff
@@ -2,6 +2,15 @@
...
+! leanprover-community/mathlib commit c4c2ed622f43768eff32608d4a0f8a6cec1c047d
```
Comment (semorrison) [inline Mathlib/Data/HashMap.lean:7]:
```suggestion
! leanprover-community/mathlib commit 2220b0cbab795e73674b8191170b0cc68c6b54a8
```

File: Mathlib/Data/HashMap.lean (hunk 1 above)
Comment (semorrison) [inline Mathlib/Data/HashMap.lean:6]:
```suggestion
! This file was ported from Lean 3 source module data.hash_map
```

**Comment by semorrison** [comment]
bors merge

File: Mathlib/Data/HashMap.lean [hunk 3]
```diff
@@ -2,6 +2,15 @@
...
+! Please do not edit these lines, except to modify the commit id
+! if you have ported upstream changes.
+
+As `HashMap` has been completely reimplemented in `Std`,
+nothing from the mathlib3 file `data.hash_map` is reflected here.
+The porting header is just here to mark that no further work on `data.hash_map` is desired.
```
Comment (eric-wieser) [inline Mathlib/Data/HashMap.lean:13]: Can we at least `#align` the old and new hash map, with something like `#align hash_map Std.HashMapₓ`?

**Comment by eric-wieser** [comment]
bors merge

**Pull request closed without merging**